### PR TITLE
feat(minio): install minio only if the storage type is minio

### PR DESCRIPTION
**Pull request by kmala**

### CODE REVIEW

File: workflow-dev/tpl/objectstorage.toml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#helm:generate helm template -o $HELM_GENERATE_DIR/manifests/deis-objectstorage-secret.yaml -d $HELM_GENERATE_FILE $HELM_GENERATE_DIR/tpl/deis-objectstorage-secret.yaml
```
Comment (bacongobbler) [inline workflow-dev/tpl/objectstorage.toml]: don't we still want to keep this? I still think we want to generate `manifests/deis-objectstorage-secret.yaml` when `helm generate` is called.

File: workflow-dev/tpl/deis-objectstorage-secret.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#helm:generate helm template -o $HELM_GENERATE_DIR/manifests/deis-objectstorage-secret.yaml -d $HELM_GENERATE_DIR/tpl/objectstorage.toml $HELM_GENERATE_DIR/tpl/deis-objectstorage-secret.yaml
```
Comment (bacongobbler) [inline workflow-dev/tpl/deis-objectstorage-secret.yaml]: Ah I see it was moved here. Disregard.

**Comment by kmala** [comment]
requires https://github.com/helm/helm/pull/439

**Comment by arschles** [comment]
Requires https://github.com/deis/minio/pull/90

**Comment by arschles** [comment]
@kmala tested this with GCS and got the following cluster status:

``` console
ENG000656:workflow-manager-api aaronschlesinger$ kd get po -w 
NAME                          READY     STATUS             RESTARTS   AGE
deis-builder-02xp7            1/1       Running            0          9m
deis-controller-23cmv         0/1       CrashLoopBackOff   7          9m
deis-database-sr5al           1/1       Running            0          9m
deis-logger-c3pi0             1/1       Running            0          9m
deis-logger-fluentd-9ho8e     1/1       Running            0          9m
deis-logger-fluentd-av0j8     1/1       Running            0          9m
deis-logger-fluentd-m0rku     1/1       Running            0          9m
deis-logger-fluentd-mi8vp     1/1       Running            0          9m
deis-logger-fluentd-vna9f     1/1       Running            0          9m
deis-registry-63lvg           1/1       Running            0          9m
deis-router-tx0zg             1/1       Running            0          9m
deis-workflow-manager-y7teq   1/1       Running            0          9m
```

The controller logs show this:

``` console
ENG000656:workflow-manager-api aaronschlesinger$ kd logs -f deis-controller-23cmv
system information:
Django Version: 1.9.5
Python 3.5.1
Django checks:
System check identified no issues (2 silenced).
Health Checks:
Checking if database is alive
There was a problem connecting to the database
```

And, since the controller logs indicate a connection problem to the DB, the DB logs are in this gist: https://gist.github.com/arschles/dd6a2411cddd26982a39b78e59f23235

Not sure if this is related to your PR at all, though...

**Comment by kmala** [comment]
this issue is because the db password changed during the helm generate and trying to recover using the new password

**Comment by arschles** [comment]
ok, so it seems that issue isn't related to your patch then ...

**Comment by mboersma** [comment]
I can't force `helm generate` to run again after this chart change, even with `--force`:

```
$ helm generate -x manifests -f workflow-dev
---> Ran 14 generators.
$ helm generate -x manifests -f workflow-dev
[ERROR] File /Users/matt/.helm/workspace/charts/workflow-dev/manifests/deis-minio-rc.yaml already exists. To overwrite it, please re-run this command with the --force/-f flag.
---> Ran 14 generators.
```

**Comment by kmala** [comment]
@mboersma created a PR in helm to fix the issue...https://github.com/helm/helm/pull/444

**Comment by kmala** [comment]
punting to beta4 as it depends on change on helm side

**Comment by kmala** [comment]
this will work now as the helm side fix is merged and released.....i want someone to test it and LGTM so i can merge it

**Comment by jchauncey** [comment]
@kmala anything in particular to test here? and what version of helm do i need

**Comment by kmala** [comment]
@jchauncey  you can the latest helm classic....when you install workflow-dev with this PR if the storage is not minio....minio pod should not be created

**Comment by jchauncey** [comment]
@kmala this is ready to merge

**Comment by kmala** [comment]
i am doing one final round of testing by downloading the latest helmc using the script

**Pull request merged**